src/Cli/dotnet/commands/dotnet-new/OptionalWorkloadProvider.cs

```diff
@@ -35,8 +36,9 @@ public Task<IReadOnlyList<ITemplatePackage>> GetAllTemplatePackagesAsync(Cancell
             var sdkDirectory = Path.GetDirectoryName(typeof(DotnetFiles).Assembly.Location);
             var sdkVersion = Path.GetFileName(sdkDirectory);
             var dotnetRootPath = Path.GetDirectoryName(Path.GetDirectoryName(sdkDirectory));
+            var userHome = CliFolderPathCalculator.DotnetHomePath;
 
-            var packages = optionalWorkloadLocator.GetDotnetSdkTemplatePackages(sdkVersion, dotnetRootPath);
+            var packages = optionalWorkloadLocator.GetDotnetSdkTemplatePackages(sdkVersion, dotnetRootPath, userHome);
```

Comment:
Is there a reason to pass in the `userHome` all the way down through the call stack?  Instead, could the `SdkDirectoryWorkloadManifestProvider` and `WorkloadResolver` have the logic in them to calculate the user home folder?

If we are passing this value around, I'd suggest passing the dotnet user home (`CliFolderPathCalculator.DotnetUserProfileFolderPath`) around instead of just the user home without `.dotnet` in it.

Reply:
> Is there a reason to pass in the userHome all the way down through the call stack? Instead, could the SdkDirectoryWorkloadManifestProvider and WorkloadResolver have the logic in them to calculate the user home folder?

I'm passing it down to be able to set its value when running tests.

> If we are passing this value around, I'd suggest passing the dotnet user home (CliFolderPathCalculator.DotnetUserProfileFolderPath) around instead of just the user home without .dotnet in it.

I'll change it to that.